marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/builtins/steps/mapping/entity-services/lib.sjs

```diff
@@ -476,21 +476,31 @@ function getMarkLogicMappingFunctions() {
   return fn.head(datahub.hubUtils.queryLatest(function() {
     let fnMetadata = fn.collection("http://marklogic.com/entity-services/function-metadata")
     let ns = {"m":"http://marklogic.com/entity-services/mapping"};
+    const functionsAdded = [];
     let output = [];
 
     for (const metaData of fnMetadata){
       if(metaData.xpath("/m:function-defs",ns)) {
         let j = 1;
         let fnLocation = metaData.xpath("/m:function-defs/@location",ns)
         for (const mlFunction of metaData.xpath("/m:function-defs/m:function-def",ns )){
-          let funcName = metaData.xpath("/m:function-defs/m:function-def["+j+"]/@name", ns);
+          let funcName = String(metaData.xpath("/m:function-defs/m:function-def["+j+"]/@name", ns));
           let params = String(metaData.xpath("/m:function-defs/m:function-def["+j+"]/m:parameters/m:parameter/@name",ns)).replace("\n",",");
           j++;
-          let singleFunction ={};
-          singleFunction["functionName"] = funcName;
-          singleFunction["signature"] = funcName +"("+params+")";
-          singleFunction["category"] = (String(fnLocation).includes("/data-hub/5/mapping-functions")) ? "builtin" : "custom";
-          output.push(singleFunction)
+          if(functionsAdded.includes(funcName)){
+            let addedFunction = output.find(func => {
+              return func.functionName == funcName
+            });
+            addedFunction["signature"] = funcName +"("+params+")";
```

Comment:
If there's a test failing for this, can the test be upgraded to verify that there's only one instance of the "echo" function? And this seems to be introducing logic that says - if there are duplicates, then whichever one is found last "wins" - i.e. the signature is for the last one. Is that the case?

Reply:
The functionality is going to remain the same. Earlier the response was an object, so the last one always wins. 
```
{"echo":
"category":"custom"
"signature":"echo(value1, value2)"}
```
Now since it is an array, it can look like the below array, so I will have to include the logic.
```
[
{
"functionName": "echo,
"category":"custom",
"signature": "echo(value)"
},
{
"functionName": "echo,
"category":"custom",
"signature": "echo(value1, value2)"
}
]
```

 I will add a test that checks that echo (if present as the function was addd by previous tests) is present only once.